specs/Pills.md

```diff
@@ -0,0 +1,188 @@
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be draggable  */
+  draggable?: boolean;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissAction?: ShorthandValue<AlertDismissActionProps>;
+}
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pills>
+  <Pill />
+</Pills>
+```
+
+- _**Internal**_
+
+```jsx
+<ElementType >
+  <li>
+    <span>
+      {children}
+    </span>
+  <li>
+</ElementType>
+```
+
+- _**DOM**_
+
+```html
+<ul aria-label>
+  <li role="presentation">
+    <span role="button">
+      ...
+    </span>
+  </li>
+</ul>
+```
+
+## Migration
+
+- _Migration from v8_
+
+...
+
+- _Migration from v0_
+
+`Dropdown` component should use `Pills` over `DropdownSelectedItem`.
+
+### Pill with dismiss
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Tab key to set focus on the Pill
+- Tab key moves focus to next item
+- Esc key remove focus
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
+| Enter/Delete       | OnDelete | The pill is removed since there is only one primary action         |
+| Backspace/Ctrl + X | OnDelete | Backspace/Ctrl + X                                                 |
+
+### Pill with popup
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Popup opens after 500ms of mouse hover state
+- Clicking in the Pill open popup and clicking outside should dimiss it
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| Esc                | OnFocus  | Dismis popup                                                       |
+| Enter              | OnFocus  | Open Popup                                                         |
+| Backspace/Ctrl + X | OnDelete | Backspace/Ctrl + X                                                 |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
```

Comment:
confused by `aria-describedby`, shouldn't it take an id of a DOM element ?

Reply:
Yes it should https://www.w3.org/TR/wai-aria/#aria-describedby ;)

Reply:
Yeah, it should.